modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/MultiSelectTag.java

```diff
@@ -0,0 +1,237 @@
+import com.liferay.portal.kernel.util.Validator;
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Set;
+
+import javax.servlet.jsp.JspException;
+import javax.servlet.jsp.JspWriter;
+
+/**
+ * @author Kresimir Coko
+ */
+public class MultiSelectTag extends BaseContainerTag {
+
+	@Override
+	public int doStartTag() throws JspException {
+		setAttributeNamespace(_ATTRIBUTE_NAMESPACE);
+
+		return super.doStartTag();
+	}
+
+	public String getClearAllTitle() {
+		return _clearAllTitle;
+	}
+
+	public String getCloseButtonAriaLabel() {
+		return _closeButtonAriaLabel;
+	}
+
+	public String getInputName() {
+		return _inputName;
+	}
+
+	public String getInputValue() {
+		return _inputValue;
+	}
+
+	public Object getLocator() {
+		return _locator;
+	}
+
+	public ArrayList<Object> getSelectedItems() {
+		return _selectedItems;
+	}
+
+	public ArrayList<Object> getSourceItems() {
+		return _sourceItems;
+	}
+
+	public boolean isDisabled() {
+		return _disabled;
+	}
+
+	public boolean isDisabledClearAll() {
+		return _disabledClearAll;
+	}
+
+	public boolean isValid() {
+		return _valid;
+	}
+
+	public void setClearAllTitle(String clearAllTitle) {
+		_clearAllTitle = clearAllTitle;
+	}
+
+	public void setCloseButtonAriaLabel(String closeButtonAriaLabel) {
+		_closeButtonAriaLabel = closeButtonAriaLabel;
+	}
+
+	public void setDisabled(boolean disabled) {
+		_disabled = disabled;
+	}
+
+	public void setDisabledClearAll(boolean disabledClearAll) {
+		_disabledClearAll = disabledClearAll;
+	}
+
+	public void setInputName(String inputName) {
+		_inputName = inputName;
+	}
+
+	public void setInputValue(String inputValue) {
+		_inputValue = inputValue;
+	}
+
+	public void setIsValid(boolean valid) {
+		_valid = valid;
+	}
+
+	/**
+	 * @deprecated As of Cavanaugh (7.4.x), replaced by {@link
+	 *             #setLocator(Object)}
+	 */
+	@Deprecated
+	public void setLabelLocator() {
+	}
+
+	public void setLocator(Object locator) {
+		_locator = locator;
+	}
+
+	public void setSelectedItems(ArrayList<Object> selectedItems) {
+		_selectedItems = selectedItems;
+	}
+
+	public void setSourceItems(ArrayList<Object> sourceItems) {
+		_sourceItems = sourceItems;
+	}
+
+	/**
+	 * @deprecated As of Cavanaugh (7.4.x), replaced by {@link
+	 *             #setLocator(Object)}
+	 */
+	@Deprecated
+	public void setValueLocator() {
+	}
+
+	@Override
+	protected void cleanUp() {
+		super.cleanUp();
+
+		_clearAllTitle = null;
+		_closeButtonAriaLabel = null;
+		_disabled = false;
+		_disabledClearAll = false;
+		_inputName = null;
+		_inputValue = null;
+		_locator = null;
+		_selectedItems = null;
+		_sourceItems = null;
+		_valid = false;
+	}
+
+	@Override
+	protected String getHydratedModuleName() {
+		return "frontend-taglib-clay/MultiSelect";
+	}
+
+	@Override
+	protected Map<String, Object> prepareProps(Map<String, Object> props) {
+		props.put("clearAllTitle", _clearAllTitle);
```

Comment:
If an attribute is not defined we will pass `null` here, which is not the same as `undefined` for js. So to prevent failures on this context we can just set those props that have a default value in Clay when they're defined. So this would look like:

```
	@Override
	protected Map<String, Object> prepareProps(Map<String, Object> props) {
		if (Validator.isNotNull(_clearAllTitle)) {
			props.put("clearAllTitle", _clearAllTitle);
		}

		if (Validator.isNotNull(_closeButtonAriaLabel)) {
			props.put("closeButtonAriaLabel", _closeButtonAriaLabel);
		}

		props.put("disabled", _disabled);
		props.put("disabledClearAll", _disabledClearAll);
		props.put("inputName", _inputName);

		if (Validator.isNotNull(_inputValue)) {
			props.put("inputValue", _inputValue);
		}

		props.put("valid", _valid);

		if (_selectedItems != null) {
			props.put("selectedItems", _selectedItems);
		}

		if (_locator != null) {
			props.put("locator", _locator);
		}

		if (_sourceItems != null) {
			props.put("sourceItems", _sourceItems);
		}

		return super.prepareProps(props);
	}
```

Reply:
Are we using`!= null` instead of `Validator.isNotNull()` because `Validator.isNotNull()` doesn't work on objects?

Such a simple solution, not sure why I didn't think of it, I think it's because I didn't expect the need for that in `prepareProps`.

Reply:
The other approach is to set in the tag the same default values as the component (which is probably better for SSR puroposes), so you don't need to do this.